src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -3,52 +3,47 @@
 import entities.Category;
 import entities.EntityException;
 import entities.MonthlyStorage;
-import views.add_edit_category_views.CategoryP;
 
 import java.util.ArrayList;
 import java.util.NoSuchElementException;
 import java.util.Objects;
 
 /**
- * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category. (Updates MonthlyStorage?)
+ * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category.
+ * Updates MonthlyStorage:
+ * 1. adding a new Category to MonthlyStorage.categoryData with MonthlyStorage.addCategory()
+ * 2.
  * Contains a helper method isValidDouble() and findCategory().
  */
 
 public class CategoryUCI implements CategoryIB {
     final CategoryOB categoryOB;
-    private CategoryOD categoryOD;
 
     /**
      * Constructs CategoryUCI.
      * @param categoryP presenter that is related to the use case.
      */
 
-    public CategoryUCI(CategoryP categoryP) {
+    public CategoryUCI(CategoryOB categoryP) {
         this.categoryOB = categoryP;
-        this.categoryOD = null;
     }
 
     /**
-     * Helper method returns a boolean to check if the input Object is a valid double.
+     * Helper method returns a double to check if the input Object is a valid double.
      * @param value a user input
```

Comment:
I believe the new toDouble just converts to double instead of checking if its a double. Also, does it not throw exception if the input isn't valid nuber?

Reply:
I don't see how it would throw a InputMismatchException? Since every object can be turned into a string, the only thing that could throw is the Double.parseDouble method, but that one should throw a NumberFormatException.

Reply:
Nevermind, yeah, you are right. She assigns the result to a double variable, but the NumberFormatException will be caught before she does that.